services/leaderboard/src/main/java/io/openliberty/spacerover/leaderboard/rest/Leaderboard.java

```diff
@@ -108,12 +105,19 @@ public Response add(LeaderboardEntry entry) {
 	@Operation(summary = "List the leaderboard from the database.")
 	@GET
 	@Path("/")
-	public Response retrieve() {
+	public Response retrieve(@DefaultValue("0") @QueryParam("startTime") String startTime,
+			@DefaultValue("") @QueryParam("endTime") String endTime) {
 		try {
+			if(endTime.isEmpty())
+			{
+				endTime = Long.toString(System.currentTimeMillis());
+			}
 			MongoCollection<Document> collection = db.getCollection(LEADERBOARD_COLLECTION_NAME);
```

Comment:
can we set the default end time in the @DefaultValue annotation instead?

```suggestion
	public Response retrieve(@DefaultValue("0") @QueryParam("startTime") String startTime,
			@DefaultValue(Long.toString(System.currentTimeMillis())) @QueryParam("endTime") String endTime) {
		try {
			MongoCollection<Document> collection = db.getCollection(LEADERBOARD_COLLECTION_NAME);
```

Reply:
unfortunately you can't, it must be a constant expression.